### DISCUSSION TITLE
Grid height mapping gives error (min 4 points) any way I try to configure..?

### DISCUSSION THREAD

**Discussion by andymid3D** [terjeio/ioSender#390] [Q&A] [answered]
Hi,
when trying grid mapping, I get always an error saying min 4 points required, however I am testing with X-200 x Y500 area, 50mm grid size.. on pressing start the head goes to relative X-200 Y500 from starting point (work 0), but then gives this error and cannot find a combination that will continue... AFAIK, a X200 grid at 50mm should have 5 points, and Y500 should have 11 points..? so total 55 points..? Why do I get this min 4 points error..?

Thanks!

Andy
:-)

**Comment by terjeio** [comment, marked as answer]
Width (W) or height (H) is 0? This works for me:

![image](https://github.com/terjeio/ioSender/assets/20260062/6762080c-2a41-4929-8929-3f956467f8e1)

**Reply by andymid3D** [reply]
So one point of the grid has to be 0..? I will also try this on Wednesday when in front of the machine! 
I feel that the Area to probe box is somewhat complicated to understand still.. don't see why if X is -200, W(idth) is 200..? should it not be -200..? And anyway, in this test, I am trying to do a grid map from the bottom right corner, when bottom left corner is X0 Y-2500 (X home is minimum, Y home is maximum), so really am trying to do from X1300, Y-2500 (G92 X0 Y0), a grid of X-200 Y500..? Or am I thinking about this the wrong way round..? Should I be starting from X1100 Y-2500, and doing the map until X1300 Y-2000..? 
Is pretty confusing this... once I understand it better I will try and express it like a noob would understand ;-)
or maybe a graphical indication of what will happen would be clearer... although understand that adding this type of GUI may stretch resources, or be more work than is worth, but until the grid map is done, that place where it will appear would ideal for a "preview" of the map...🤔

**Reply by andymid3D** [reply]
Hi Terje,
So today I could test this case, and I think I can confirm that the W and H fields cannot have negative numbers..? When I replicated your example, all worked fine, so I did another example starting from X0 and W going -200, and it showed the same error about min 4 points.. thereafter tried a few other combinations (one also by mistake X0 W0: cannot have infinite area or something similar!), and basically I see that the mapping grid points must go +X and +Y directions...? 🧐 W & H must be positive although the grid can start from anywhere with respect to Wpos0,0..?

For my client use case, it would be nice to be able to do -X, as he always uses XmaxYmin as his job starting point... having to start the grid from the other side of the work piece is not a real problem, only would be more in line with his work orientation, FWIW ;-) 

Also, from another post I've seen around here, would be nice to have some numerical data about the points measured, so far have only seen the color gradient red (highest points) to blue (lowest points), but to have an idea about how much difference there is by clicking on different points would be cool! No rush thou, just getting the mapping going and being able to apply to a job is great! The rest is icing on the cake 👍

Cheers!

Andy
:-)

**Reply by phil-barrett** [reply]
You can save your height map in a file and look at it via a text editor.  All the height offsets are there.

**Reply by andymid3D** [reply]
> You can save your height map in a file and look at it via a text editor. All the height offsets are there.

OK! very useful info, thanks! was unaware... I hadn't gotten round to saving the height map today, I presume the file name ending is irrelevant, it will open in a text editor..? (lazy question really, will find out next time I try it out 😜)
